src/main/scala/is/hail/utils/TextTableReader.scala

```diff
@@ -13,15 +13,23 @@ import org.apache.spark.sql.Row
 object TextTableReader {
 
   def splitLine(s: String, separator: String, quote: java.lang.Character): Array[String] = {
-    val p = Pattern.compile(separator)
-    val m = p.matcher(s)
 
-    def matchSep(i: Int): Int = {
-      m.region(i, s.length)
-      if (m.lookingAt())
-        m.end() - m.start()
-      else
-        -1
+    val matchSep: Int => Int = separator.length match {
+      case 0 => fatal("Hail does not currently support 0-character separators")
+      case 1 =>
+        val sepChar = separator(0)
+        (i: Int) => if (s(i) == sepChar) 1 else -1
+      case _ =>
+        val p = Pattern.compile(separator)
+        val m = p.matcher(s)
+
+      { (i: Int) =>
```

Comment:
Indentation seems wrong?  Sometimes I find IntelliJ gets confused when  { starts a line.  You might try putting the { after the =>.